client/my-sites/earn/memberships/products.tsx

```diff
@@ -178,12 +200,13 @@ function MembershipsProductsSection( { query }: MembersProductsSectionProps ) {
 										{ currentProduct?.title }
 									</div>
 									<sub className="memberships__products-product-price">
-										{ currentProduct.subscribe_as_site_subscriber
-											? translate( '%s/month', { args: price } )
-											: price }
-										{ currentAnnualProduct && translate( ' (%s/year)', { args: annualPrice } ) }
+										{ getPriceFromProduct( currentProduct, price ) }
+										{ currentAnnualProduct &&
+											translate( ' (%s)', {
+												args: getPriceFromProduct( currentAnnualProduct, annualPrice ),
+											} ) }
```

Comment:
This code all works, but I'm trying to understand why lines 204-207 are needed. Wouldn't the new getPriceFromProduct() method return the annual price already when invoked on line 203? Or to put it differently, seems like we could potentially output two price lines - one from line 203, and one frm line 204-7.

Reply:
Those are needed for tiers: The prices you see in the following screenshot is for 2 different products ( `currentProduct` and `currentAnnualProduct`): 

<img width="310" alt="Screenshot 2023-11-08 at 16 14 09" src="https://github.com/Automattic/wp-calypso/assets/790558/8ea80948-f852-4313-85e3-8c281cd755dc">

Reply:
Ah, makes sense. Thanks!